### PR TITLE
#167165016 Set up Code Climate

### PR DESCRIPTION
**Pull request by VictorMaria**
#### What does this PR do?
Set up Code Climate

#### Description of Task to be completed?
Set up Code Climate in Circle CI config file
Add Repo to Code Climate
Add test coverage and maintainability badge to README.md

#### How should this be manually tested?
N/A

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#167165016](https://www.pivotaltracker.com/story/show/167165016)

#### Screenshots (if appropriate)
Screen Shot 2019-08-27 at 6 05 58 PM

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
     "start": "node .",
     "build": "NODE_ENV=production webpack --config ./config/webpack.prod.js",
     "lint": "npx eslint ./src/**.js --fix",
-    "test": "jest",
+    "test": "jest --coverage",
```
Comment (tolulope-od) [inline package.json:10]: Please, why is a coverage flag added to this script here again? There's a script that collects coverage already

**Review by tolulope-od — COMMENTED** [review]
Great job adding tests for the previously created components!

File: package.json (hunk 1 above)
Comment (ocranbillions) [reply package.json:10]: I don't think we'll need two separate scripts for test and coverage. What he can do is leave the test script as it is and take out the test:coverage. If that's fine.

File: .circleci/config.yml [hunk 2]
```diff
@@ -5,6 +5,8 @@
 version: 2
 jobs:
   build:
+    environment:
+      CC_TEST_REPORTER_ID: d5a3c72fe28b01b8c3194ff55140594f31b8c77524c3b927b97ad24085f2fe36
```
Comment (mazma1) [inline .circleci/config.yml]: Must this be a hardcoded value?

File: .eslintrc [hunk 3]
```diff
@@ -6,7 +6,13 @@
...
       "es6": true,
       "jest": true,
-      "cypress/globals": true
+      "cypress/globals": true,
+    },
+    "globals": {
+      "mount": true,
+      "render": true,
+      "toJson": true,
+      "shallow": true,
```
Comment (mazma1) [inline .eslintrc:15]: Why do we have these?

**Comment by mazma1** [comment]
Why do you have 'unrelated' file changes for this chore?

File: package.json (hunk 1 above)
Comment (VictorMaria) [reply package.json:10]: I have taken out the test:coverage script. Thank you for your observations.

File: .eslintrc (hunk 3 above)
Comment (VictorMaria) [reply .eslintrc:15]: I used mount and toJson for testing the components. The other two will be needed for further testing as we introduce more codes to the repo, that is why I included them.

File: .circleci/config.yml (hunk 2 above)
Comment (VictorMaria) [reply .circleci/config.yml]: No, I will fix it.

**Comment by VictorMaria** [comment]
@mazma1 The unrelated file changes are as a result of the tests I had to write for two components and a reducer function. The lines in those codes were not covered and there is a test coverage constraint on the repo that makes Circle CI fail if the codes don't meet the specified coverage percentage.

**Pull request merged**